sympy/vector/tests/test_vector.py

```diff
@@ -28,6 +30,14 @@ def test_cross():
     assert Cross(v1, v2).doit() == C.z**3*C.i + (-C.x*C.z)*C.j + (C.x*C.y - C.x*C.z**2)*C.k
     assert cross(v1, v2) == C.z**3*C.i + (-C.x*C.z)*C.j + (C.x*C.y - C.x*C.z**2)*C.k
     assert Cross(v1, v2) == -Cross(v2, v1)
+    # XXX: Cannot use Cross here. See XFAIL test below:
+    assert cross(v1, v2) + cross(v2, v1) == Vector.zero
+
+
+@XFAIL
+def test_cross_xfail():
+    v1 = C.x * i + C.z * C.z * j
+    v2 = C.x * i + C.y * j + C.z * k
     assert Cross(v1, v2) + Cross(v2, v1) == Vector.zero
```

Comment:
Note that the behaviour on master is broken:
```python
In [1]: from sympy.vector import CoordSys3D, Cross
   ...: N = CoordSys3D('N')

In [2]: v = Cross(N.i, N.j)

In [3]: v
Out[3]: (i_N)×(j_N)

In [4]: v + v
Out[4]: 0
```
The test here only passes on master because all Cross terms are dropped from a VectorAdd. With this PR there will be an error instead of silently incorrect results.

It is not simple to fix Cross because the whole idea of an unevaluated `Cross` goes against the basic design of vector which is that a vector is represented as a components dict:
```python
In [7]: (2*N.i + N.j).components
Out[7]: {i_N: 2, j_N: 1}
```

Reply:
```
from sympy import *
from sympy.vector import Cross, CoordSys3D

A = CoordSys3D('A')
Cross(A.x, A.z) + Cross(A.z, A.x)
```
```
---> 59 return self._components

AttributeError: 'Cross' object has no attribute '_components'
```

I don't expect the automatic simplification, but adding two `Cross` gets into error.

Reply:
It should have always given an error except that buggy code was added to ignore this case.

A future PR can make it actually work. Every Vector method needs the components attribute e.g. on master:
```python
In [1]: from sympy import *
   ...: from sympy.vector import Cross, CoordSys3D
   ...: 
   ...: A = CoordSys3D('A')

In [2]: Cross(x*A.i, A.j).diff(x)
...
AttributeError: 'Cross' object has no attribute '_components'
```
Having vector expressions that are not represented this way requires deeper changes than just adding a `Cross` class. There is no way to represent the sum of two Cross expressions as a VectorAdd.

Reply:
I'm okay with the change. 
However, since I'm not familar with this module, I'd wait for relevant people to approve the merge. Maybe @Upabjojr can check about this.